Sync v2-10-stable with v2-10-test(sync_v2_10_test) to release 2.10.3

utkarsharma2 opened the pull request:
Time for 2.10.3rc1!

vincbeck committed:
Fix documentation for cpu and memory usage (#42147) (#42256)

These metrics are reported in airflow/task/task_runner/standard_task_runner.py without "percent"

(cherry picked from commit 3f6497b528a9ca50c363f80b74025046aaec4215)

Co-authored-by: lucasmo <lucas.madar@gmail.com>

romsharon98 committed:
fix: Correct docstring format in _get_template_context (#42244) (#42272)

(cherry picked from commit 8fe286108720ba1f9a303c066022349d6656f735)

Co-authored-by: Wonseok Yang <deadline.yang@gmail.com>

pierrejeambrun committed:
Bump dompurify from 2.2.9 to 2.5.6 in /airflow/www (#42263) (#42270)

Bumps [dompurify](https://github.com/cure53/DOMPurify) from 2.2.9 to 2.5.6.
- [Release notes](https://github.com/cure53/DOMPurify/releases)
- [Commits](https://github.com/cure53/DOMPurify/compare/2.2.9...2.5.6)

---
updated-dependencies:
- dependency-name: dompurify
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
(cherry picked from commit fc5f686e3a238b7e11034057e62524ce9f598b80)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jscheffl committed:
Update StatsD Image Tag from failed dependencies check (#42264) (#42281)

(cherry picked from commit b5576c081f00e376faea1b27459430166ac6975b)

vincbeck committed:
Use `selectinload` in trigger (#40487) (#42351)

(cherry picked from commit 46b41e332f2d32962c0e3c14c1e5b8d052dfc359)

Co-authored-by: Joseph Ang <josephangbc@gmail.com>

romsharon98 committed:
apply otel_service on metrics (#42242) (#42441)

* apply otel_service on metrics

* change version_added to 2.10.3

(cherry picked from commit e14b4cae9c61cbf1df6e6b573d926d614f9f63e5)

jscheffl committed:
Bugfix task execution from runner in Windows (#42426) (#42478)

(cherry picked from commit 737900a0a66d6987c5931dee34a634e32c812490)

shahar1 committed:
Fix the span link of task instance to point to the correct span in the scheduler_job_loop (#42430) (#42480)

(cherry picked from commit ae6ed8a9480ec18b2b14804821c1a4e9f1a91f3c)

Co-authored-by: Howard Yoo <32691630+howardyoo@users.noreply.github.com>

jscheffl committed:
Do not attempt to provide not stringified objects to UI via xcom if pickling is active (#42388) (#42486)

* Do not attempt to provide not stringified objects to UI via xcom if pickling is active

* Add pytest

(cherry picked from commit f9877af256acd1b64fcf91aefdf14ab9389b0bbc)

pierrejeambrun committed:
fix: ensure DAG trigger form submits with updated parameters upon keyboard submit (#42487) (#42499)

(cherry picked from commit 5af74639503df4cc723e5663b4abb99ba8022147)

Co-authored-by: Danny Liu <dannyjialiliu@gmail.com>

pierrejeambrun committed:
Handle ENTER key correctly in trigger form and allow manual JSON (#42525) (#42535)

(cherry picked from commit 3ef29a641a852c076da7a9611c325669e8485368)

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

pierrejeambrun committed:
Fix broken main re generated api typescript comment (#42500) (#42537)

(cherry picked from commit 73dd6c17bd10ddda63a1682ac2174b0d206590dd)

Co-authored-by: Daniel Standish <15932138+dstandish@users.noreply.github.com>

pierrejeambrun committed:
reduce eyestrain in dark mode with reduced contrast and saturation (#42567) (#42583)

* reduce eyestrain in dark mode with reduced contrast and saturation

* feat: readjusted saturation

---------

Co-authored-by: Curtis Bangert <bangert.curtis+git@gmail.com>
(cherry picked from commit 97c7d2c23d73985b2f7956160f00b879b90ef6d8)

Co-authored-by: codecae <codecae@users.noreply.github.com>

potiuk committed:
Doc update - Airflow local settings no longer importable from dags folder (#42231) (#42603)

---------

Co-authored-by: Jishan Garg <jigarg@expediagroup.com>
(cherry picked from commit f4f38f15fe4e6c32614686ceb3b5991dd91ef7ee)

Co-authored-by: JISHAN GARGACHARYA <34843832+jishangarg@users.noreply.github.com>

potiuk committed:
Support of host.name in OTEL metrics and usage of OTEL_RESOURCE_ATTRIBUTES in metrics (#42428) (#42604)

* fixes: 42425, and 42424

* fixed static type check failure

(cherry picked from commit db06cb8e893addeef4d49479beb1f2387fa63993)

Co-authored-by: Howard Yoo <32691630+howardyoo@users.noreply.github.com>

jscheffl committed:
Prevent redirect loop on /home with tags/lastrun filters (#42607) (#42609) (#42628)

Closes #42607

(cherry picked from commit be2e48ef41327c86474e91b4283826bf2c27ebfc)

Co-authored-by: Julian Maicher <jmaicher@users.noreply.github.com>

shahar1 committed:
[BACKPORT] Add retry logic in the scheduler for updating trigger timeouts in case of deadlocks. (#41429) (#42651)

* Add retry logic in the scheduler for updating trigger timeouts in case of deadlocks. (#41429)

* Add retry in update trigger timeout

* add ut for these cases

* use OperationalError in ut to describe deadlock scenarios

* [MINOR] add newsfragment for this PR

* [MINOR] refactor UT for mypy check

(cherry picked from commit 00589cf8fe8faffa6f994e9b85717cb2babc1631)

* Fix type-ignore comment for typing changes (#42656)

---------

Co-authored-by: TakawaAkirayo <153728772+TakawaAkirayo@users.noreply.github.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

bbovenzi committed:
Correctly select task in DAG Graph View when clicking on its name (#38782) (#42697)

* Fix in DAG Graph View, clicking Task on it's name doesn't select the task. (#37932)

* Updated TaskName onClick

* Fixed missing onToggleCollapse

* Added missing changes

* Updated: rebase

* fixed providers error message

* undo fab changes

* Update user_command.py

---------

Co-authored-by: jonhspyro <121674572+jonhspyro@users.noreply.github.com>

Lee-W committed:
fix(datasets/managers): fix error handling file loc when dataset alias resolved into new datasets (#42733)

* fix(datasets/managers): fix error handling fileloc when datasetalias resolved into new datasets

* test(datasets/manager): add test case test_register_dataset_change_with_alias

* refactor(datasets/manager): simplify for loop

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

pierrejeambrun committed:
Fix dag warning documentation (#42858) (#42888)

(cherry picked from commit b7c8ebb7d9f0d5a3d1598667371ab69582b4e66a)

jscheffl committed:
Improving validation of task retries to handle None values (#42532) (#42915)

* Improving validation of task retries to handle None values

* Updated the validation check to use the "or" logical operator

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

* Added a comment for the new validation check

* Update baseoperator.py

* Update taskinstance.py

---------

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>
(cherry picked from commit 1114ab2d03c4c58a68f0a75f03c82a39385c60d0)

Co-authored-by: sonu4578 <veetech4455@gmail.com>

kaxil committed:
Remove the referrer from Webserver to Scarf (#42901) (#42942)

This will make sure we don't receive any information about the Webserver URL sending the info like the number of plugins and such.

(cherry picked from commit 39f8e1d487ab8f1969aaec512fdffacca989813c)

potiuk committed:
Restrict looker-sdk version 24.18.0 and microsoft-kiota-http 1.3.4 (#42954) (#42977)

* restrict looker version 24.18.0

* update microsoft provider deps with microsoft-kiota-http

* update gh issue ids to provider yaml files

* ignore type in applfowhook

* ignore type in applfowhook

* ignore type in applfowhook

(cherry picked from commit 7d8ea68bdaae3258bd391b8f6ae0277258a7c437)

Co-authored-by: GPK <gopidesupavan@gmail.com>

infused-kim committed:
Docs: Add templating info to TaskFlow tutorial (#42992)

(cherry picked from commit 437616df18d0be3644a88d24c684b50afe3933ba)

jason810496 committed:
Fix PythonOperator when DAG has hyphen in name (#42993)

gopidesupavan committed:
manually commiting cherry pick 6da1799 (#42983)

pierrejeambrun committed:
Update json schema pre-commit to have draft7 schema in file (#43005) (#43007)

We request this file each time the CI is running pre-commit and sometimes
it fails with 403 error. This PR adds the draft7 schema file locally for use
by pre-commit to avoid this error

(cherry picked from commit 5316e618e0953563b81875677a21b23749817173)

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

ephraimbuddy committed:
Flush the session between writing and deletion of RTIF (#42928) (#43012)

* FLush the session before deleting the RTIF data

Previously, this was how it was done, but now,
a session was used for both the writing and deletion of RTIF,
which we suspect caused StaleDataError. The related PR: https://github.com/apache/airflow/pull/38565

This PR brings back the old behaviour of using different sessions for writing/deleting RTIFs

* fixup! Use different sessions in writing and deletion of RTIF

* add test and use flush

(cherry picked from commit ced319fe95a731b745801fe9b15ca7b24ef0e82f)

gopidesupavan committed:
Disable flaky mssql based integration tests (#42811) (#43016)

* quarantined flaky mssql integration tests

* disable mssql from ci integration checks

potiuk committed:
Removed unicodecsv dependency for providers with Airflow version 2.8.0 and above (#42765) (#42970)

(cherry picked from commit 00b8452b7526166b2918976b991aff1124d1a17c)

Co-authored-by: Gopal Dirisala <39794726+dirrao@users.noreply.github.com>

gopidesupavan committed:
update k8s tests urllib3 retry config status_forcelist and allowed_methods (#42871) (#43026)

gopidesupavan committed:
Improve startup of K8S tests (#42721) (#43025)

gopidesupavan committed:
increase backoff_factor and add try/catch in k8s tests (#42940) (#43030)

vincbeck committed:
Deprecate session auth backend (#42911)

jscheffl committed:
Add retry on error 502 and 504 (#42994) (#43044)

* Add retry on error 502 and 504

* fix mypy findings

* Add pytest

* Convert response code to HTTPStatus

* Add docs to retriable exception

* extend docs for AirflowHttpException

* Fix syntax and typos

* fix pytest

* fix static checks

* fix some static checks

* Fix ruff

* fix pre-commit

---------

Co-authored-by: Majoros Donat (XC-DX/EET2-Bp) <donat.majoros2@hu.bosch.com>
(cherry picked from commit 477d74718adb9b3d68621910af061891b9bdcc17)

Co-authored-by: majorosdonat <mjd2bp@bosch.com>

jscheffl committed:
Fix v2-10 build dependencies (#43052)

uranusjr committed:
[Backport] Remove zombie from executor (#43065)

potiuk committed:
Clarifying PLUGINS_FOLDER permissions by DAG authors (#43022) (#43029)

(cherry picked from commit c471c31111958f0a4dde775f559d5c606f3149a8)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

jscheffl committed:
Added task_instance_mutation_hook for mapped operator index 0 (#42661) (#43089)

* Added task_instance_mutation_hook for mapped operator index 0

* Added unit test

---------

Co-authored-by: Marco Küttelwesch <marco.kuettelwesch@de.bosch.com>
(cherry picked from commit b7007e2b146e6ef929a211925a3d4397b1e9955d)

Co-authored-by: AutomationDev85 <96178949+AutomationDev85@users.noreply.github.com>

jscheffl committed:
AIP-69: Breeze adjustments for introduction of Edge Executor (#41731) (#43139)

* Breeze adjustments for introduction of AIP-69 Remote Executor

* Rename Remote Executor to Edge Executor

(cherry picked from commit 8b15840b94c88f6ed16036baacb974459f7865ec)

uranusjr committed:
Always use logical date in DAG run header (#43148) (#43163)

jscheffl committed:
Update trove-classifiers in v2-10 test, backport PR #43021 (#43194)

* Update trove-classifiers in v2-10 test

* Back-Port Add upperbound to microsoft-kiota-abstractions (#43021)

jscheffl committed:
Update trove-classifiers to 2024.10.21.16

(cherry picked from commit fa28e3595d2caa2d8ae266fc4740d8673a09877b)

jscheffl committed:
Temporarily limits WTForms to below 3.2.0 (#43233) (#43234)

WTForms are limited to 3.2.0 because of the error in tests. We technically do not need it directly
as this is a dependency of Flask-WTF, but we need to specify it here to add the limitation
The issue to track it is https://github.com/pallets-eco/wtforms/issues/863
Note. 3.2.0 has been broken because of imports https://github.com/pallets-eco/wtforms/issues/861 which
was fixed in 3.2.1, but after import was fixed, the tests started to work with 3.2.1
when the issue 863 is fixed, we should likely leave the line below and specify !=3.2.0,!=3.2.1

(cherry picked from commit 979d750b226fccbc5dac174f5b9ac566ec663e56)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

potiuk committed:
Do not fail the build if only trove-classifiers change (#43236) (#43237)

Trove classifiers is a non-code package so generally spaeking we
have no need to update it every time it is released, but it would
be great if we could update it together with other build
dependencies - and not fail the build when only trove classifiers
change, but update trove-classifiers nevertheless when we run

pre-commit run --hook-stage manual update-build-dependencies --all-files

This is very similar to what we already do with UV.

This PR achieves that:

* in CI we set the env SKIP_TROVE_CLASSIFIERS_ONLY to true
* this one will check if the only change in pyproject.toml are
  trove-classifiers
* If so - it only prints warning and exits with success

potiuk committed:
Skip example importability tests for providers in non-main branches (#43260) (#43263)

When "Always" tests are running in v2* branches, they also tested
provider examples for importability. However sometimes those tests
failed because we have not cherry-picked some of the provider's
changes that were necessary for the examples to get imported.

This PR excludes provider's examples from the "always" running
importability tests, when the branch we are running it in is
not main (i.e. we are in one of the past v2-branches).

(cherry picked from commit 0f38be1f957d72dd8b6081e7f2381f82f513f78a)

potiuk committed:
Better handling masking of values of set variable  (#43123) (#43278)

(cherry picked from commit 1260d9c042f7a6351a4b67611ce39f91764dcbdf)

Co-authored-by: Shubham Raj <48172486+shubhamraj-git@users.noreply.github.com>

potiuk committed:
Fixes behaviour of example dag tests for main/other branches (#43273) (#43307)

While the #43260 attempted to address the problem where example
dag importability tests should skip provider tests on non-main,
it did not actually solve the problem.

While debugging it, it turned out that since #42505, the provider
tests were not executed in main "at all" - the "providers"
directory was not included in the list of places to check for
the example dags (they were in "airflow" in v2-10-test") this is
why it "looked like" the solution worked in "main".

This PR fixes both problems:

* brings back importability of provider's example_dags in main branch
* properly excludes the providers examples in non-main branch

potiuk committed:
Fix edge-case when conflicting constraints prevent k8s env creation (… (#43298)

* Fix edge-case when conflicting constraints prevent k8s env creation (#43276)

The #36930 added constraints to creation of k8s environment, but it
had a side effect - the constraints could not be created if source
of airflow had dependencies conflicting with constraints (which
might happen for example when we upgrade FAB - because locally
pinned FAB might be different than the one in constraints).

Also the constraints were "hard-coded" - the python version,
branch and github repository were hard-coded.

This PR fixes both problems:

* constraints URL is dynamically generated based on current
  branch, repo and python version
* in case attempts to create the venv with constraints fails,
  we attempt to install it again without constraints

(cherry picked from commit 274b6e1168f84561e8f652a1a4a6fc82aeadc477)

* Update k8s_requirements.txt

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,55 @@
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+- Add ``run_with_db_retries`` when the scheduler updates the deferred Task as failed to tolerate database deadlock issues. (#41429)
```
This should just be a bugfix. It's not noteworthy to end users really.

utkarsharma2 replied:
I too think so, but wasn't sure.

utkarsharma2 replied:
Moved it to bug-fix section, PTAL.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,55 @@
+
+Miscellaneous
+"""""""""""""
+- Temporarily limits WTForms to below 3.2.0 (#43233) (#43234)
```
We should wait for #43309 and remove this one.

potiuk replied:
Agree - if we are to **just** remove the description - but cherry-pick all commits regardless. It's just easier to apply them all :D

kaxil replied:
Backport PR: https://github.com/apache/airflow/pull/43318

kaxil replied:
that PR is merged, so feel free to adjust the release notes now @utkarsharma2

utkarsharma2 replied:
Updated the release notes post cherry-picking the commits. PTAL.

jedcunningham reviewed (commented):
The list could use a "does the title make sense for an end user" pass or 2, but overall looks good.

potiuk committed:
Add isolation mode exclusion for mapped operator test (#43297) (#43311)

This test has been cherry-picked to v2-10-test and should be
excluded from isolation mode.

(cherry picked from commit f4d9a1b068ab3fed4e527bd8dbc67043caae620e)

potiuk committed:
Check python version that was used to install pre-commit venvs (#43282) (#43310)

Since we moved to Python 3.9 and started usign 3.9+ only features
line functools.cache, having Python 3.8 as default python version
will result in virtualenvs created with outdated python.

This pre-commit checks what is the global version of python3 and it
will complain if it is lower than 3.9. This is done via explicit
call of "python3", because this is what pre-commit does. Also
global setting is needed because otherwise new virtualenvs
created by pre-commit might be created using lower python version.

(cherry picked from commit 076da924696dc7d8814a38f442e9ca3eb3e90634)

potiuk committed:
fix schedule_downstream_tasks bug (#42582) (#43299)

* fix schedule_downstream_tasks bug

* remove partial_subset

* Update comment

---------

Co-authored-by: 维湘 <jiazhao.ljz@alibaba-inc.com>
(cherry picked from commit 3fceaa69260be80bc2123cd4664db79d96142b9f)

Co-authored-by: luoyuliuyin <luoyuliuyin@gmail.com>

potiuk committed:
Mark sometimes failing heartbeat test and view test as flaky (#43250) (#43257)

(cherry picked from commit fcfc6f96df8e3e22364952dc035b46dce840974f)

potiuk committed:
Upgrade FAB to 4.5.1 (#43251) (#43316)

FAB 4.5.1 has been released in September with a few small
fixes. This change updates fab to 4.5.1 including changing the
rate limiter creation that is vendored in. It has been changed
in https://github.com/dpgaspar/Flask-AppBuilder/pull/2254 and
relased in 4.5.1. That's the only dfference in security manager
between 4.5.0 and 4.5.1.

(cherry picked from commit 84ff10bf06cf1a529169990d25c00a33d06e740e)

kaxil committed:
Backport: Bump Flask-AppBuilder to ``4.5.2`` (#43309) (#43318)

shahar1 committed:
Fix instruction for docker compose(tested on Mac M1) (#43119) (#43321)

* Fix instruction for docker compose(tested on Mac M1)

Make fixes that allows to run airflow dags in pycharm debugger on Mac M1

* fix static checks

* fix linter error

* fix unexpected identation

---------

Co-authored-by: Shahar Epstein <60007259+shahar1@users.noreply.github.com>
(cherry picked from commit d725cf683ed25955f739909d7176899224dca13a)

Co-authored-by: Bohdan Udovenko <bohdan.udov@gmail.com>

Joffreybvn committed:
[Backport] BashOperator: Execute templated bash script as file (#43191)



---------

Co-authored-by: Joffrey Bienvenu <joffrey.bienvenu@infrabel.be>

potiuk committed:
Masking configuration values irrelevant to DAG author (#43040) (#43336)

Some configurations are irrelevant to DAG authors and hence we need to mask those to avoid it from getting logged unknowingly.

Co-authored-by: adesai <adesai@cloudera.com>
Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>
(cherry picked from commit 0b030c562363dd924bbbee0793636be18deeabe3)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

potiuk committed:
Suppress warnings when masking sensitive confs (#43335) (#43337)

This is to prevent issues such as https://github.com/apache/airflow/pull/43334

(cherry picked from commit 0a9c3c007bf1672398a63bb9c97b82ec48f60afc)

---------

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>
Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

shahar1 committed:
Fixed type in section Object Storage XCom Backend in file xcoms.rst (#43339) (#43342)

(cherry picked from commit 7c6c106591f745b254a119f09da0ebc0729c9d98)

Co-authored-by: Baitur Ulukbekov <ulukbekovbr@gmail.com>
(cherry picked from commit a13c49eea8f85ebc6847dc3f1bf32f6618877cc5)

utkarsharma2 committed:
Update Airflow version to 2.10.3

utkarsharma2 committed:
Update RELEASE_NOTES.rst

kaxil committed:
Add Airflow 3 ui & task_sdk directories to gitignore (#43319)

(cherry picked from commit 0c6c8508ff234cb70292b75d927ba2e5c41c5110)

kaxil committed:
Remove Scarf analytics from Airflow Webserver (#43346) (#43348)

Since we have already made decisions about what we want to do for Plugins in Airflow 3 and revamping the entire UI, this data isn't that important. There were also concerns raised about de-depulications and other things.

(cherry picked from commit dc25301f35e78f5058bca9eb8436ec31dc3a2962)
(cherry picked from commit 2860f81d7e0e822531ebc90f6c9b4882f8ae82de)

Lee-W committed:
ci(pre-commit): migrate pre-commit config (#43372)

We update the names of the pre-commit stages. From the https://pre-commit.com/#confining-hooks-to-run-at-certain-stages:

new in 3.2.0: The values of stages match the hook names. Previously, commit, push, and merge-commit matched pre-commit, pre-push, and pre-merge-commit respectively.

(cherry picked from commit add8e9d1a37279d1755138756a194d1dad693f93)

Lee-W committed:
rename ill-named constraints in dag_schedule_dataset_alias_reference table #43314

(cherry picked from commit 9a868e089a0d7723423867b28fadab273a197e18)

romsharon98 committed:
prevent out of index error in getDagRunLabel function (#43388) (#43404)

(cherry picked from commit 4652fed1c64136b4e8f6f85ca46f7b9f92ab5caf)
